lib/import/brca/providers/nottingham/nottingham_handler.rb

```diff
@@ -37,94 +86,149 @@ def process_fields(record)
+            process_cdna_or_exonic_variants(genotype, record)
+            process_protein_impact(genotype, record)
             process_varpathclass(genotype, record)
             add_organisationcode_testresult(genotype)
-            extract_teststatus(genotype, record) # added by Francesco
+            assign_test_status(record, genotype)
             @persister.integrate_and_store(genotype)
           end
 
           def add_organisationcode_testresult(genotype)
             genotype.attribute_map['organisationcode_testresult'] = '698A0'
           end
 
-          def add_simple_fields(genotype, record)
+          def add_moleculartestingtype(record, genotype)
             testingtype = record.raw_fields['moleculartestingtype']
-            genotype.add_molecular_testing_type_strict(TEST_TYPE_MAP[testingtype.downcase.strip])
-            # variant_path_class = record.raw_fields['teststatus']
-            # genotype.add_variant_class(variant_path_class.downcase) unless variant_path_class.nil?
-            received_date = record.raw_fields['sample received in lab date']
-            genotype.add_received_date(received_date.downcase) unless received_date.nil?
-          end
-
-          def add_complex_fields(genotype, record)
-            Maybe(record.raw_fields['disease']).each do |disease|
-              case disease.downcase.strip
-              when 'hereditary breast and ovarian cancer (brca1/brca2)'
-                genotype.add_test_scope(:full_screen)
-              when 'brca1/brca2 pst'
-                genotype.add_test_scope(:targeted_mutation)
-              end
+            genotype.add_molecular_testing_type_strict(TEST_TYPE_MAP[testingtype])
+          end
+
+          def assign_test_scope(record, genotype)
+            testscopefield = record.raw_fields['disease']
+            testtypefield = record.raw_fields['moleculartestingtype']
+            if TEST_SCOPE_MAP[testscopefield].present?
+              genotype.add_test_scope(TEST_SCOPE_MAP[testscopefield])
+            elsif %w[PALB2 CDH1 TP53].include? testscopefield
+              genotype.add_test_scope(TEST_SCOPE_TTYPE_MAP[testtypefield])
             end
-            # Maybe(record.raw_fields['genotype']).each do |geno|
-            #   @genotype_counter += 1
-            #   @failed_genotype_parse_counter += genotype.add_typed_location(@ex.extract_type(geno))
-            # end
           end
 
-          def extract_teststatus(genotype, record)
-            case record.raw_fields['teststatus'].to_s.downcase
-            when /normal|completed/i
+          def normaltest_nullvariantfield?(record, teststatusfield, variantfield)
+            return false if record.raw_fields['teststatus'].nil?
+
+            teststatusfield == 'Normal' && variantfield.blank?
+          end
+
+          def normaltest_controlvariantfield?(record, teststatusfield, variantfield)
+            return false if record.raw_fields['teststatus'].nil? &&
```

Comment:
@FrancescoNea , you already have teststatusfield and variantfield available in the method so conditions can be checked on them rather accessing them again via record.raw_fields. Same for other methods as well. Thanks